syntax = "proto3";

package transport_catalogue_serialize;

import "graph.proto";

message RoutingSettings {
    uint32 bus_wait_time = 1;
    double bus_velocity = 2;
}

message StopInfo {
    string stop_name = 1;
    uint32 edge_id = 2;
}

message BusInfo {
    uint32 edge_id = 1;
    string bus_name = 2;
}

message CountInfo {
    uint32 edge_id = 1;
    uint32 stop_count = 2;
}

message TransportRouter {
    RoutingSettings routing_settings = 1;
    repeated StopInfo stop_info = 2;
    repeated BusInfo bus_info = 3;
    repeated CountInfo count_info = 4;
    repeated Edge edges = 5;
}

